v0/destinations/posthog/transform.js

```diff
@@ -0,0 +1,137 @@
+const { EventType } = require("../../../constants");
+const {
+  DEFAULT_BASE_ENDPOINT,
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG
+} = require("./config");
+const {
+  defaultRequestConfig,
+  getBrowserInfo,
+  getDeviceModel,
+  constructPayload,
+  defaultPostRequestConfig,
+  ErrorMessage
+} = require("../../util");
+
+const isValidUrl = url => {
+  try {
+    // eslint-disable-next-line no-new
+    new URL(url);
+    return true;
+  } catch (err) {
+    return false;
+  }
+};
+
+// Logic To match destination Property key that is in Rudder Stack Properties Object.
+const generatePropertyDefination = message => {
+  const PHPropertyJson = CONFIG_CATEGORIES.PROPERTY.name;
+  let propertyJson = MAPPING_CONFIG[PHPropertyJson];
+  let data = {};
+
+  if (message.channel === "mobile") {
+    propertyJson = propertyJson.filter(d => {
+      return d.isScreen || d.all;
+    });
+  } else {
+    propertyJson = propertyJson.filter(d => {
+      return !d.isScreen || d.all;
+    });
+  }
+
+  if (
+    message.channel === "web" &&
+    message.context &&
+    message.context.userAgent
+  ) {
+    const browser = getBrowserInfo(message.context.userAgent);
+    const osInfo = getDeviceModel(message);
+    data.$os = osInfo;
+    data.$browser = browser.name;
+    data.$browser_version = browser.version;
+  }
+  data = {
+    ...constructPayload(message, propertyJson),
+    ...data
+  };
+
+  if (message.type === EventType.SCREEN) {
+    data.$screen_name = message.event;
+  }
+
+  if (isValidUrl(data.$current_url) && !data.$host) {
+    const url = new URL(data.$current_url);
+    data.$host = url.host;
+  }
+
+  return data;
+};
+
+const responseBuilderSimple = (message, category, destination) => {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  if (!payload) {
+    // fail-safety for developer error
+    // eslint-disable-next-line no-undef
+    throw Error(ErrorMessage.FailedToConstructPayload);
+  }
+
+  payload.properties = {
+    ...generatePropertyDefination(message),
+    ...payload.properties
+  };
+
+  // Mapping Destination Event with correct value
+  if (category.type !== CONFIG_CATEGORIES.TRACK.type) {
+    payload.event = category.event;
+  }
+
+  const responseBody = {
+    ...payload,
+    api_key: destination.Config.teamApiKey,
+    type: category.type
+  };
+  const response = defaultRequestConfig();
+  response.endpoint = destination.Config.yourInstance
+    ? `${destination.Config.yourInstance}/batch`
```

Comment:
Need to check for trailing `/` otherwise, we'll be adding 2 `/`s

Reply:
Fixed